app/tools/update/updater.js

```diff
@@ -0,0 +1,45 @@
+// native modules
+const EventEmitter = require('events').EventEmitter;
+// 3rd party modules
+const _ = require('lodash');
+const Promise = require('bluebird');
+// application modules
+const Npm = require('./npm');
+
+
+class Updater extends EventEmitter {
+  constructor(cwd = __dirname, env = _.pick(process.env, ['PWD', 'PATH', 'HOME'])) {
+    super();
+    this._options = {cwd, env};
+    this._pending = Promise.resolve();
+  }
+  update(revision) {
+    if (this._pending.isPending()) {
+      return Promise.reject('Updating in progress.');
+    }
+    this._pending = this._update(revision).catch((error) => {
```

Comment:
dummy promise to fetch current deployed revision (to allow reverting back..) (implementation not needed in this PR)